"Not a label multiset" error when persisting canvas

The following exception is thrown when trying to persist the canvas because `isLabelMultiset` attribute is missing in downsampled label datasets:
```
java.io.IOException: /volumes/segmentation/multicut/data/s1 is not a label multiset dataset.
    at org.janelia.saalfeldlab.n5.imglib2.N5LabelMultisets.openLabelMultiset(N5LabelMultisets.java:120)
    at org.janelia.saalfeldlab.n5.imglib2.N5LabelMultisets.openLabelMultiset(N5LabelMultisets.java:100)
    at org.janelia.saalfeldlab.n5.imglib2.N5LabelMultisets.openLabelMultiset(N5LabelMultisets.java:83)
    at org.janelia.saalfeldlab.n5.imglib2.N5LabelMultisets.openLabelMultiset(N5LabelMultisets.java:66)
    at org.janelia.saalfeldlab.paintera.data.n5.CommitCanvasN5.downsampleAndWriteBlocksLabelMultisetType(CommitCanvasN5.java:660)
    at org.janelia.saalfeldlab.paintera.data.n5.CommitCanvasN5.persistCanvas(CommitCanvasN5.java:272)
    at org.janelia.saalfeldlab.paintera.data.mask.MaskedSource.lambda$persistCanvas$13(MaskedSource.java:606)
    at java.lang.Thread.run(Thread.java:748)
Exception in thread "Thread-63" java.lang.RuntimeException: Error while trying to persist.
    at org.janelia.saalfeldlab.paintera.data.mask.MaskedSource.lambda$persistCanvas$13(MaskedSource.java:621)
    at java.lang.Thread.run(Thread.java:748)
Caused by: org.janelia.saalfeldlab.paintera.data.mask.persist.UnableToPersistCanvas: Unable to commit canvas.
    at org.janelia.saalfeldlab.paintera.data.n5.CommitCanvasN5.persistCanvas(CommitCanvasN5.java:304)
    at org.janelia.saalfeldlab.paintera.data.mask.MaskedSource.lambda$persistCanvas$13(MaskedSource.java:606)
    ... 1 more
Caused by: java.io.IOException: /volumes/segmentation/multicut/data/s1 is not a label multiset dataset.
    at org.janelia.saalfeldlab.n5.imglib2.N5LabelMultisets.openLabelMultiset(N5LabelMultisets.java:120)
    at org.janelia.saalfeldlab.n5.imglib2.N5LabelMultisets.openLabelMultiset(N5LabelMultisets.java:100)
    at org.janelia.saalfeldlab.n5.imglib2.N5LabelMultisets.openLabelMultiset(N5LabelMultisets.java:83)
    at org.janelia.saalfeldlab.n5.imglib2.N5LabelMultisets.openLabelMultiset(N5LabelMultisets.java:66)
    at org.janelia.saalfeldlab.paintera.data.n5.CommitCanvasN5.downsampleAndWriteBlocksLabelMultisetType(CommitCanvasN5.java:660)
    at org.janelia.saalfeldlab.paintera.data.n5.CommitCanvasN5.persistCanvas(CommitCanvasN5.java:272)
    ... 2 more
```

This started happening after https://github.com/saalfeldlab/paintera/pull/281, where I switched to `N5LabelMutlisets` for reading label mutliset data, which requires `isLabelMutliset` attribute as it doesn't differentiate between full-res and downsampled datsets.

--- a/src/main/java/org/janelia/saalfeldlab/paintera/data/n5/CommitCanvasN5.java
+++ b/src/main/java/org/janelia/saalfeldlab/paintera/data/n5/CommitCanvasN5.java
@@ -608,6 +608,7 @@ public class CommitCanvasN5 implements PersistCanvas
 		return t;
 	}
 
+	// TODO: switch to N5LabelMultisets for writing label multiset data
 	private static void writeBlocksLabelMultisetType(
 			final RandomAccessibleInterval<UnsignedLongType> canvas,
 			final long[] blocks,
@@ -644,6 +645,7 @@ public class CommitCanvasN5 implements PersistCanvas
 		}
 	}
 
+	// TODO: switch to N5LabelMultisets for writing label multiset data
 	private static void downsampleAndWriteBlocksLabelMultisetType(
 			final long[] affectedBlocks,
 			final N5Writer n5,

--- a/src/main/java/org/janelia/saalfeldlab/paintera/data/n5/CommitCanvasN5.java
+++ b/src/main/java/org/janelia/saalfeldlab/paintera/data/n5/CommitCanvasN5.java
@@ -657,6 +657,11 @@ public class CommitCanvasN5 implements PersistCanvas
 			final TLongObjectHashMap<BlockDiff> blockDiffsAt
 			) throws IOException {
 
+		// In older converted data the "isLabelMultiset" attribute may not be present in s1,s2,... datasets.
+		// Make sure the attribute is set to avoid "is not a label multiset" exception.
+		n5.setAttribute(previousDataset.dataset, N5Helpers.IS_LABEL_MULTISET_KEY, true);
+		n5.setAttribute(targetDataset.dataset, N5Helpers.IS_LABEL_MULTISET_KEY, true);
+
 		final RandomAccessibleInterval<LabelMultisetType>  previousData = N5LabelMultisets.openLabelMultiset(n5, previousDataset.dataset);
 
 		for (final long targetBlock : affectedBlocks)

--- a/src/main/java/org/janelia/saalfeldlab/paintera/ui/dialogs/create/CreateDataset.java
+++ b/src/main/java/org/janelia/saalfeldlab/paintera/ui/dialogs/create/CreateDataset.java
@@ -201,7 +201,7 @@ public class CreateDataset
 
 				if (name == null || name.equals(""))
 					throw new IOException("Name not specified!");
-				N5Data.createEmptyLabeLDataset(
+				N5Data.createEmptyLabelDataset(
 						container,
 						dataset,
 						dimensions.getAs(new long[3]),

--- a/src/main/java/org/janelia/saalfeldlab/util/n5/N5Data.java
+++ b/src/main/java/org/janelia/saalfeldlab/util/n5/N5Data.java
@@ -471,6 +471,7 @@ public class N5Data {
 		return openLabelMultiset(reader, dataset, transform, globalCache, priority);
 	}
 
+	// TODO: switch to N5LabelMultisets for reading label multiset data. Currently it is not possible because of using a global cache.
 	public static ImagesWithInvalidate<LabelMultisetType, VolatileLabelMultisetType> openLabelMultiset(
 			final N5Reader reader,
 			final String dataset,

--- a/src/main/java/org/janelia/saalfeldlab/util/n5/N5Data.java
+++ b/src/main/java/org/janelia/saalfeldlab/util/n5/N5Data.java
@@ -673,7 +673,7 @@ public class N5Data {
 	 * @param maxNumEntries limit number of entries in each {@link LabelMultiset} (set to less than or equal to zero for unbounded)
 	 * @throws IOException if any n5 operation throws {@link IOException}
 	 */
-	public static void createEmptyLabeLDataset(
+	public static void createEmptyLabelDataset(
 			final String container,
 			final String group,
 			final long[] dimensions,
@@ -683,7 +683,7 @@ public class N5Data {
 			final double[][] relativeScaleFactors,
 			final int[] maxNumEntries) throws IOException
 	{
-		createEmptyLabeLDataset(container, group, dimensions, blockSize, resolution, offset,relativeScaleFactors, maxNumEntries, false);
+		createEmptyLabelDataset(container, group, dimensions, blockSize, resolution, offset,relativeScaleFactors, maxNumEntries, false);
 	}
 
 	/**
@@ -701,7 +701,7 @@ public class N5Data {
 	 * @throws IOException if any n5 operation throws {@link IOException} or {@code group}
 	 * already exists and {@code ignorExisting} is {@code false}
 	 */
-	public static void createEmptyLabeLDataset(
+	public static void createEmptyLabelDataset(
 			final String container,
 			final String group,
 			final long[] dimensions,
@@ -765,11 +765,12 @@ public class N5Data {
 			n5.createDataset(dataset, scaledDimensions, blockSize, DataType.UINT8, new GzipCompression());
 			n5.createDataset(uniqeLabelsDataset, scaledDimensions, blockSize, DataType.UINT64, new GzipCompression());
 
-			// {"maxNumEntries":-1,"compression":{"type":"gzip","level":-1},"downsamplingFactors":[2.0,2.0,1.0],"blockSize":[64,64,64],"dataType":"uint8","dimensions":[625,625,125]}%
+			// {"maxNumEntries":-1,"compression":{"type":"gzip","level":-1},"downsamplingFactors":[2.0,2.0,1.0],"blockSize":[64,64,64],"dataType":"uint8","dimensions":[625,625,125],
+			// "isLabelMultiset":true}%
 			n5.setAttribute(dataset, N5Helpers.MAX_NUM_ENTRIES_KEY, maxNum);
-			if (scaleLevel == 0)
-				n5.setAttribute(dataset, N5Helpers.IS_LABEL_MULTISET_KEY, true);
-			else
+			n5.setAttribute(dataset, N5Helpers.IS_LABEL_MULTISET_KEY, true);
+
+			if (scaleLevel != 0)
 			{
 				n5.setAttribute(dataset, N5Helpers.DOWNSAMPLING_FACTORS_KEY, accumulatedFactors);
 				n5.setAttribute(uniqeLabelsDataset, N5Helpers.DOWNSAMPLING_FACTORS_KEY, accumulatedFactors);

